rewrite-maven/src/test/java/org/openrewrite/maven/RemoveRedundantDependencyVersionsTest.java

```diff
@@ -1530,4 +1530,50 @@ void keepVersionIfManagedIsNonExact() {
           )
         );
     }
+
+    @Test
+    @Issue("https://github.com/openrewrite/rewrite/discussions/4386")
+    void dontOverrideDependencyConfigurations() {
+        rewriteRun(
+          spec -> spec.recipe(new RemoveRedundantDependencyVersions(null, null, RemoveRedundantDependencyVersions.Comparator.GTE, null)),
+          pomXml(
+            """
+                <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+                    <modelVersion>4.0.0</modelVersion>
+                    <parent>
+                        <groupId>org.springframework.boot</groupId>
+                        <artifactId>spring-boot-starter-parent</artifactId>
+                        <version>2.3.12.RELEASE</version>
+                    </parent>
+                    <groupId>com.example</groupId>
+                    <artifactId>acme</artifactId>
+                    <version>0.0.1-SNAPSHOT</version>
+                    <dependencyManagement>
+                        <dependencies>
+                            <dependency>
+                                <groupId>org.springframework.boot</groupId>
+                                <artifactId>spring-boot-starter-web</artifactId>
+                                <!-- The version number should be dropped -->
```

Comment:
This line was a silly left over; turns out we can't (for as much as I know) leave out a version on a managed dependency. At least our internals complain. How's that for you @crankydillo ?
```suggestion
```

Reply:
You are right!

`mvn validate` failed complaining about missing version.  I guess I'm misreading the XSD or something.. 

Hmm, so what to do.  Functionality wise, my guess is we'd prefer the dependency to have a version that matches any newly changed upstream dependencyMgmt section.

Reply:
Then I'd for now lean towards doing the safe thing, which is what I've implemented here, as to not make any change at all just yet. That's already an improvement over removing a known wrong case. I propose we merge this as-is and iterate in a follow up PR on a better approach if we can find one that's feasible. Would you agree?

Reply:
My initial reaction was I'd prefer to leave it (opt 3), but I am starting to change my mind:). I'd be OK merging.

I understand this conversation needs to happen under new-approach PR, but just to set the stage..  I can think of 3 states:

1. Remove (current behavior)
2. Leave dependency with version set to old version (your current proposal).
3. Leave but set version to new managed dependency version (what I proposed above)

I don't think is 'the answer'.  I lean towards configuration.

Reply:
I do agree that option 3 would be best; but perhaps also best tackled separately as it needs to be worked in. Option 2 is at least better than option 1, so then let's merge and iterate. :) Thanks again for the provided tests & helping think through the options!